xarray/core/computation.py

```diff
@@ -974,27 +977,30 @@ def dot(*arrays, **kwargs):
         dims = [dims]
 
     common_dims = set(arrays[0].dims)
+    all_dims = []
```

Comment:
would it work to make `all_dims` a set instead of a list? I think that would be slightly more efficient.